extensions/kafka-streams/deployment/src/main/java/io/quarkus/kafka/streams/deployment/KafkaStreamsHotReplacementSetup.java

```diff
@@ -25,22 +26,19 @@ public void setupHotDeployment(HotReplacementContext context) {
 
         @Override
         public void run() {
-            if (nextUpdate < System.currentTimeMillis()) {
-                synchronized (this) {
-                    if (nextUpdate < System.currentTimeMillis()) {
-                        executor.execute(() -> {
-                            try {
-                                context.doScan(true);
-                            } catch (RuntimeException e) {
-                                throw e;
-                            } catch (Exception e) {
-                                throw new RuntimeException(e);
-                            }
-                        });
-                        // we update at most once every 2s
-                        nextUpdate = System.currentTimeMillis() + TWO_SECONDS;
+            final long now = System.currentTimeMillis();
+            // The getAndUpdate we will update the nextUpdate field if it's value is more that now then will return the current value.
```

Comment:
it's=its. The entire sentence sounds a bit odd, something seems wrong with wording/grammar.

Reply:
I believe it should be something like:

```
Check if we need to trigger the scan. The scan is triggered if the nextUpdate is greater than the current time (now).
nextUpdate is also updated if required to now + 2 seconds
```

Or something like this.

Reply:
This was just a comment targeting the next developer that will look at this code and wondering what it does :)

Maybe I need more to explain why it is like that then what it does, the trick here is that the same condition will enter the loop and update the AtomicLong atomically while protecting multiple thread doing it concurrently ...

So maybe the best comment will look like this
```
Check if we need to trigger the scan. 
The AtomicLong will protect the check from threads doing it concurrently.
The scan is triggered if the nextUpdate is greater than the current time (now). 
As the same condition is used inside the LongUnaryOperator that update the nexUpdate, when the condition is true, it will also be updated to now + 2 seconds.
```